tests/unit/cluster/cli.tcl

```diff
@@ -81,6 +81,28 @@ start_multiple_servers 3 [list overrides $base_conf] {
 
     set node1_rd [redis_deferring_client 0]
 
+    test "use previous hostip in \"cluster-preferred-endpoint-type unknown-endpoint\" mode" {
+        
+        # backup and set cluster-preferred-endpoint-type unknown-endpoint
+        $node1_rd CONFIG GET cluster-preferred-endpoint-type
+        set endpoint_type_before_set [lindex [split [$node1_rd read] " "] 1]
+        $node1_rd CONFIG SET cluster-preferred-endpoint-type unknown-endpoint
```

Comment:
No need to use the deferred client which requires a separate read after sending the command. We can use the normal client $node1 instead. It returns the reply immediately. It is simpler imo.